spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitTemplate.java

```diff
@@ -1703,33 +1710,42 @@ public Boolean isMandatoryFor(final Message message) {
 
 	private <T> T doExecute(ChannelCallback<T> action, ConnectionFactory connectionFactory) {
 		Assert.notNull(action, "Callback object must not be null");
-		Channel channel;
+		Channel channel = null;
+		boolean invokeScope = false;
+		if (this.activeTemplateCallbacks.get() > 0) {
+			channel = this.dedicatedChannels.get();
```

Comment:
This was just an attempt at optimization - only access the `ThreadLocal` if we know for sure it might contain a value. In most cases, so I wanted an optimized code path.